ahnlich/ai/src/engine/store.rs

```diff
@@ -203,11 +212,18 @@ impl AIStoreHandler {
                     storeinput_type: store_input_type,
                 });
             }
-            let metadata_value: MetadataValue = store_input.clone().into();
-            store_value.insert(metadata_key.clone(), metadata_value.clone());
+            if store_original {
+                let metadata_key = &*AHNLICH_AI_RESERVED_META_KEY;
+                if store_value.contains_key(metadata_key) {
+                    return Err(AIProxyError::ReservedError(metadata_key.to_string()));
+                }
+                let metadata_value: MetadataValue = store_input.clone().into();
+                store_value.insert(metadata_key.clone(), metadata_value.clone());
+                delete_hashset.insert(metadata_value);
+            }
             output.try_push((store_input, store_value))?;
-            delete_hashset.insert(metadata_value);
         }
+        let delete_hashset = (store_original).then_some(delete_hashset);
```

Comment:
Not storing originals also means we can't update values.

If this is right, instead of removing the original completely, is it possible for us to hash the original, so we can have something to work with instead of removing completely.

Reply:
I think we can justify not being able to update the original.... the hash itself can be pretty large as well and still costs some cycles

One thing we can offer users is to document that when not storing the original, the upside is no duplication but the downside we are unable to update that particular entry and would need them to first issue an appropriate `DelPred` on their end using any appropriate metadata set